client/src/app/layout/SidebarApp/SidebarApp.tsx

```diff
@@ -8,201 +8,208 @@ import {
   NavList,
   NavExpandable,
   PageSidebarBody,
-  SelectOptionProps,
 } from "@patternfly/react-core";
 
 import { Paths } from "@app/Paths";
 import { LayoutTheme } from "../LayoutUtils";
 import { checkAccess } from "@app/utils/rbac-utils";
 import keycloak from "@app/keycloak";
 
-import { useLocalStorage } from "@migtools/lib-ui";
-import { LocalStorageKey } from "@app/Constants";
 import { FEATURES_ENABLED } from "@app/FeatureFlags";
 import { SimpleSelectBasic } from "@app/components/SimpleSelectBasic";
 import "./SidebarApp.css";
+import { ReactElement } from "react-markdown/lib/react-markdown";
+import { adminRoutes, devRoutes } from "@app/Routes";
 
-export const SidebarApp: React.FC = () => {
-  const token = keycloak.tokenParsed || undefined;
-  const userRoles = token?.realm_access?.roles || [],
-    adminAccess = checkAccess(userRoles, ["tackle-admin"]);
+type PersonaType = "ADMINISTRATION" | "MIGRATION";
 
-  const { t } = useTranslation();
-  const { search } = useLocation();
-  const history = useHistory();
-  const PersonaKey = {
-    ADMINISTRATION: "Administration",
-    MIGRATION: "Migration",
-  };
+const PersonaKey: { [key in PersonaType]: string } = {
+  ADMINISTRATION: "Administration",
+  MIGRATION: "Migration",
+};
 
-  const personaOptions: SelectOptionProps[] = [
-    { value: PersonaKey.MIGRATION, children: PersonaKey.MIGRATION },
-    adminAccess && {
-      value: PersonaKey.ADMINISTRATION,
-      children: PersonaKey.ADMINISTRATION,
-    },
-  ].filter(Boolean);
+export const SidebarApp: React.FC = () => (
+  <Switch>
+    {devRoutes.map(({ path, exact }, index) => (
+      <Route
+        path={path}
+        exact={exact}
+        key={index}
+        render={() => <MigrationSidebar />}
+      />
+    ))}
+    {adminRoutes.map(({ path, exact }, index) => (
+      <Route
+        path={path}
+        exact={exact}
+        key={index}
+        render={() => <AdminSidebar />}
+      />
+    ))}
+  </Switch>
+);
 
-  const [selectedPersona, setSelectedPersona] = useLocalStorage<string | null>({
-    key: LocalStorageKey.selectedPersona,
-    defaultValue: null,
-  });
+const PersonaSidebar: FC<{
+  children: ReactElement;
+  selectedPersona: PersonaType;
+  targetPath: Paths;
```

Comment:
What the `targetPath` should be is a bit confusing.  I understand it as meaning, "when the persona is changed, change the url to this path".

What if `PersonaKey` and `PersonaType` looked more like this:
```ts
const PersonaKey = {
  ADMINISTRATION: {
    label: "Administration",
    targetPath: Paths.general,
  },
  MIGRATION: {
    label: "Migration",
    targetPath: Paths.applications,
  },
};

type PersonaType = keyof typeof PersonaKey;
```

Then the `targetPath` for a persona would live with the persona and would not need to be a prop at all.

Also, not that I know of any plans to add a new persona or another sidebar, a structure like this would allow adding another persona pretty easily.